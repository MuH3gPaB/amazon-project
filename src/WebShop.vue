<template>
  <div id="web-shop" class="web-shop">
    <WebShopHeader></WebShopHeader>
    <RouterView></RouterView>
  </div>
</template>

<script>
import WebShopHeader from "./components/web-shop-header";

export default {
  name: 'WebShop',
  components: {
    WebShopHeader
  }
};
</script>

<style scoped>
.web-shop {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  max-width: 1200px;
}
</style>

