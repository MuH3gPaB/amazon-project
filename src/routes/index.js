import Category from "../pages/category";
import Product from "../pages/product";
import NotFound from "../pages/not-found";
import Cart from "../pages/cart";
import ThankYou from "../pages/thank-you";

const routes = [
    {
        path: "/category/:id?",
        name: "category",
        component: Category
    },
    {
        path: "/product/:id?",
        name: "product",
        component: Product
    },
    {
        path: "/cart",
        name: "cart",
        component: Cart
    },
    {
        path: "/thankYou",
        name: "thankYou",
        component: ThankYou
    },
    {
        path: "/notFound",
        name: "notFound",
        component: NotFound
    },
    {
        path: "*",
        name: "default",
        redirect: {name: "category"}
    }
];

export default routes;