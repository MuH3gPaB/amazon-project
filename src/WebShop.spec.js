import {beforeEach, describe, it, jest, expect} from "@jest/globals";
import WebShop from './WebShop';
import WebShopStore from './store';
import {createLocalVue, mount} from "vue-test-utils";
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import routes from './routes';

function mockStore(store) {
    let mockedModule = {actions: {}, getters: {}, state: {}, mutations: {}};
    Object.keys(store.actions || []).forEach(key => mockedModule.actions[key] = jest.fn());
    Object.keys(store.getters || []).forEach(key => mockedModule.getters[key] = jest.fn());
    Object.keys(store.state || []).forEach(key => mockedModule.state[key] = store.state[key]);
    Object.keys(store.mutations || []).forEach(key => mockedModule.state[key] = store.state[key]);
    return mockedModule;
}

describe('WebShop', () => {
    let localVue;
    let store;
    let wrapper;


    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        localVue.use(VueRouter);
        store = mockStore(WebShopStore);
        store.state.categories = testCategories();
        store.state.products = testProducts();
        store.getters.isCategoryExist.mockReturnValue(() => true);
        store.actions.loadProducts.mockReturnValue(() => Promise.resolve());

        wrapper = mount(WebShop, {
            localVue,
            store: new Vuex.Store(store),
            router: new VueRouter({mode: 'history', routes})
        });
    });

    it('should match snapshot', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});

function testCategories() {
    return [
        {
            id: "104994286",
            thumbnailUrl: "https://d2j6dbq0eux0bg.cloudfront.net/images/58958138/2208495045.jpg",
            name: "Шерсть"
        },
        {
            id: "104994287",
            thumbnailUrl: "https://d2j6dbq0eux0bg.cloudfront.net/images/58958138/2208509122.jpg",
            name: "Хлопок"
        }
    ];
}

function testProducts() {
    return [
        {
            id: "344134461",
            name: "ОБРАЗЕЦ. Цветочный топ",
            defaultDisplayedPriceFormatted: "1 395.00₽",
            thumbnailUrl: "https://d2j6dbq0eux0bg.cloudfront.net/default-store/jade_tank_400px.jpg"
        },
        {
            id: "344134462",
            name: "ОБРАЗЕЦ. Черный топ",
            defaultDisplayedPriceFormatted: "1 295.00₽",
            thumbnailUrl: "https://d2j6dbq0eux0bg.cloudfront.net/default-store/black_tank_400px.jpg"
        },
        {
            id: "344134466",
            name: "ОБРАЗЕЦ. Белый топ",
            defaultDisplayedPriceFormatted: "1 495.00₽",
            thumbnailUrl: "https://d2j6dbq0eux0bg.cloudfront.net/default-store/white_tank_400px.jpg"
        }
    ];
}
