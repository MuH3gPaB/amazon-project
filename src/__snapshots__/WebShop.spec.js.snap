// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebShop should match snapshot 1`] = `
<div
  class="web-shop"
  id="web-shop"
>
  <div
    class="web-shop-header"
  >
    <div
      class="header-text"
    />
     
    <div
      class="cart-shield-container"
    >
      <div
        class="cart-shield"
      >
        <img
          class="cart-icon"
          src="../images/cart-icon.png"
        />
         
        <div
          class="cart-items-count"
        >
          0
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="category"
  >
    <div
      class="category-list-container"
    >
      <div>
        <div>
          <div
            class="category-item category-item__selected"
          >
            <img
              src="https://d2j6dbq0eux0bg.cloudfront.net/images/58958138/2208495045.jpg"
            />
             
            <div>
              <h1>
                Шерсть
              </h1>
            </div>
          </div>
        </div>
        <div>
          <div
            class="category-item"
          >
            <img
              src="https://d2j6dbq0eux0bg.cloudfront.net/images/58958138/2208509122.jpg"
            />
             
            <div>
              <h1>
                Хлопок
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="product-list-container"
    >
      <!---->
    </div>
  </div>
</div>
`;
